src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java

```diff
@@ -0,0 +1,133 @@
+	private final CANSparkFlex launcherAngleMotor;
+	private final CANSparkFlex launcherHoodMotor;
+	private final RelativeEncoder launcherTopEncoder;
+	private final RelativeEncoder launcherBottomEncoder;
+	private final SparkAbsoluteEncoder launcherAngleEncoder;
+	private final SparkAbsoluteEncoder launcherHoodEncoder;
+	private final SparkPIDController launcherAnglePIDController;
+	private final SparkPIDController launcherHoodPIDController;
+
+	GenericEntry launcherSpeed =
+			Shuffleboard.getTab("Launcher")
+					.addPersistent("Launcher Speed", SPEAKER_SHOOT_SPEED)
+					.withSize(2, 1)
+					.withWidget(BuiltInWidgets.kTextView)
+					.getEntry();
+
+	GenericEntry launcherAngle =
+			Shuffleboard.getTab("Launcher")
+					.addPersistent("Launcher angle", getAngle())
+					.withSize(1, 1)
+					.withWidget(BuiltInWidgets.kTextView)
+					.getEntry();
+	// Constructor
+	public LauncherSubsystem() {
+
+		// MOTOR INSTANCE VARIBLES
+		// motors
+		launcherTopMotor = new CANSparkFlex(Hardware.LAUNCHER_TOP_MOTOR_ID, MotorType.kBrushless);
+		launcherBottomMotor = new CANSparkFlex(Hardware.LAUNCHER_BOTTOM_MOTOR_ID, MotorType.kBrushless);
+		launcherAngleMotor = new CANSparkFlex(Hardware.LAUNCHER_ANGLE_MOTOR_ID, MotorType.kBrushless);
+		launcherHoodMotor = new CANSparkFlex(Hardware.LAUNCHER_HOOD_MOTOR_ID, MotorType.kBrushless);
+		// encoders
+		launcherTopEncoder = launcherTopMotor.getEncoder();
+		launcherBottomEncoder = launcherBottomMotor.getEncoder();
+		launcherAngleEncoder = launcherAngleMotor.getAbsoluteEncoder(Type.kDutyCycle);
+		launcherHoodEncoder =
+				launcherHoodMotor.getAbsoluteEncoder(SparkAbsoluteEncoder.Type.kDutyCycle);
+		// PID controllers
+		launcherAnglePIDController = launcherAngleMotor.getPIDController();
+		launcherHoodPIDController = launcherHoodMotor.getPIDController();
+		launcherAnglePIDController.setFeedbackDevice(launcherAngleEncoder);
+		launcherHoodPIDController.setFeedbackDevice(launcherHoodEncoder);
+		configMotors();
+	}
+
+	public void configMotors() {
+		launcherTopMotor.restoreFactoryDefaults();
+		launcherBottomMotor.restoreFactoryDefaults();
+		launcherAngleMotor.restoreFactoryDefaults();
+		launcherHoodMotor.restoreFactoryDefaults();
+
+		// idle mode (wow)
+		launcherTopMotor.setIdleMode(IdleMode.kCoast);
+		launcherBottomMotor.setIdleMode(IdleMode.kCoast);
+		launcherAngleMotor.setIdleMode(IdleMode.kBrake);
+		launcherHoodMotor.setIdleMode(IdleMode.kBrake);
+		// inveritng the bottom motor lmao
+		launcherBottomMotor.setInverted(true);
+
+		// current limit
+		launcherTopMotor.setSmartCurrentLimit(20);
+		launcherBottomMotor.setSmartCurrentLimit(20);
+		launcherAngleMotor.setSmartCurrentLimit(20);
+		launcherHoodMotor.setSmartCurrentLimit(20);
+
+		launcherTopMotor.burnFlash();
+		launcherBottomMotor.burnFlash();
+		launcherAngleMotor.burnFlash();
+		launcherHoodMotor.burnFlash();
+	}
+
+	// stop specific motor method
+	public void stopMotor(CANSparkFlex motor) {
+		motor.stopMotor();
+	}
```

Comment:
Shouldn't be public, as nothing outside this class should be passing a motor as a parameter.  Since it's never used, consider deleting it, or replacing with something like `stopLauncher()` that sets the two launcher motors to 0 power.

Reply:
Agreed - though if we do keep it, we should make sure the motors aren't trying to fully stop (which could cause physical damage to the system) and are instead not contributing any more kinetic energy to the system (thinking back to 2022). If we do go down that route `stopMotor()` should work fine provided the other refactoring advice is followed.

Reply:
i went with the stopLauncher idea, ty!